### PR TITLE
security:  follow a authenticated schedule id allows you to perform operations on any other schedule tasks

### PR DESCRIPTION
**Pull request by ysjemmm**
#### What type of PR is this?
security-bug

#### What this PR does / why we need it:
Because the server only authenticates and verifies the `scheduleId`, but does not verify whether a `scheduleTaskId` belongs to the `scheduleId`, the problem that arises is that if I have a verified `scheduleId`, I can access any other scheduleTask resource.
There is some code in the logging module that only fits `obcloud_202409`, so it is included this time

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 1]
```diff
@@ -662,4 +663,15 @@ private void handleTaskResult(String jobType, TaskResult taskResult) {
         }
     }
 
+    @Override
+    public Map<String, String> getJobAttributes(Long jobId) {
+        List<JobAttributeEntity> attributeEntityList = jobAttributeRepository.findByJobId(jobId);
+        if (attributeEntityList == null) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:669]: attributeEntityList will never be null

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java [hunk 2]
```diff
@@ -95,68 +103,74 @@ public class ScheduledTaskLoggerService {
...
+                String objId = jobAttributeMap.get(attributeKey);
+                String bucketName = jobAttributeMap.get(JobAttributeKeyConstants.LOG_STORAGE_BUCKET_NAME);
+                if (ObjectUtil.isNotNull(objId) && ObjectUtil.isNotNull(bucketName)) {
+                    return cloudObjectStorageService.generateDownloadUrl(objId).toString();
                 }
+                return StrUtil.EMPTY;
             }
+            return String.format(DOWNLOAD_LOG_URL_PATTERN, scheduleId, scheduleTaskId);
+        } catch (Exception e) {
+            log.warn("get download log url failed, scheduleId={}, scheduleTaskId={}", scheduleId, scheduleTaskId, e);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java:158]: throw exception instead

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java (hunk 1 above)
Comment (ysjemmm) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:669]: I know

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**